firmware/src/app/card.cpp

```
@@ -0,0 +1,137 @@
+  saved_settings = settings;
+  song_dirty = false;
+  settings_dirty = false;
+}
+
+// A pick from the song view (§9.6, T-56): what is on screen goes back to its own
+// slot, and the slot picked either comes back from the card or, being empty, becomes
+// a copy of what is on screen — which is a save, not a change of what is playing.
+void switch_song(int slot, Model& model, const engine::Kit& kit) {
+  const int from = model.settings.song;
+  io::save_song(from, kit, staged);
+  const bool has_song = io::load_song(slot, kit, loaded) && !is_empty(loaded);
```

Clang (14.0.6)

[warning] variable 'has_song' is not initialized

(cppcoreguidelines-init-variables)